### PR TITLE
Event model and confirmation logic

### PR DESCRIPTION
**Pull request by jtkpiotr**
@michal-konarski @lkudyba @quirk0o @mhib

### CODE REVIEW

File: spec/factories/event_factory.rb [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-    sequence :name do |n|
-      "event##{n}'s name"
+    sequence :event_id do |n|
+      "event##{n}"
     end
-    description '123'
-    user 'user'
-    start_time Time.new(2016, 0o1, 0o1)
-    end_time Time.new(2016, 0o1, 0o1) + 10.minutes
     association :conference_room, factory: :conference_room
```
Comment (quirk0o) [inline spec/factories/event_factory.rb]: I think you can just write `conference_room` and that's it. Definitely don't need to specify factory if it has the same name

File: config/routes.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@
   get 'oauth2callback' => 'authentication#authenticate'
   post 'notify/:conference_room_id' => 'notification#receive', as: :notifications
   get 'conference_rooms/:conference_room_id/events' => 'events#room_index'
+  post 'events/confirm' => 'events#confirm'
```
Comment (quirk0o) [inline config/routes.rb]: Why not `conference_rooms/:conference_room_id/events/:event_id/confirm` ?

File: app/models/google_calendar/google_event.rb [hunk 3]
```diff
@@ -21,8 +21,19 @@ def find_by_room(time_interval, conference_room_ids)
...
 
+    def confirm(conference_room_id, event_id)
+      event = Event.find_by(event_id: event_id)
+      event = save_event(conference_room_id, event_id) if event.nil?
+      event.confirm
+    end
+
     private
 
+    def save_event(conference_room_id, event_id)
```
Comment (quirk0o) [inline app/models/google_calendar/google_event.rb]: `create_event` would be better :(

File: app/models/google_calendar/google_event.rb [hunk 4]
```diff
@@ -21,8 +21,19 @@ def find_by_room(time_interval, conference_room_ids)
       event_finder.by_room(time_interval, conference_room_ids)
     end
 
+    def confirm(conference_room_id, event_id)
+      event = Event.find_by(event_id: event_id)
```
Comment (quirk0o) [inline app/models/google_calendar/google_event.rb]: Maybe use `find_or_create`?

File: config/routes.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@
   get 'oauth2callback' => 'authentication#authenticate'
   post 'notify/:conference_room_id' => 'notification#receive', as: :notifications
   get 'conference_rooms/:conference_room_id/events' => 'events#room_index'
-
+  post 'conference_rooms/:conference_room_id/events/:event_id' => 'events#confirm', as: :confirmation
```
Comment (quirk0o) [inline config/routes.rb]: `.../confirm` maybe?

File: spec/models/event_spec.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    it { is_expected.to validate_uniqueness_of :event_id }
+  end
+
+  describe '#confirm' do
+    context 'event is not saved in database' do
+      let(:event) { build :event }
+
+      it 'sets confirmed field to true' do
+        event.confirm
```
Comment (mjk-old) [inline spec/models/event_spec.rb]: Put that in `before` or `subject!` since its duplicated in two places.

File: spec/models/google_event_spec.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  let(:google_event) { described_class.new(credentials, user_email) }
+
+  describe '#confirm' do
+    context 'given event_id that doest not exist in db' do
+      let(:event_id) { 'invalid_event_id' }
+
+      context 'given invalid conference_room_id' do
+        let(:conference_room_id) { 0 }
+        it 'raises ActiveRecord::RecordNotFound' do
+          expect { google_event.confirm(conference_room_id, event_id) }.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (mjk-old) [inline spec/models/google_event_spec.rb]: Ditto, move that to `subject`.

File: spec/models/google_event_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe GoogleCalendar::GoogleEvent do
+  let(:credentials) { 'credentials' }
+  let(:user_email) { 'email@example.com' }
+  let(:google_event) { described_class.new(credentials, user_email) }
+
+  describe '#confirm' do
+    context 'given event_id that doest not exist in db' do
```
Comment (mjk-old) [inline spec/models/google_event_spec.rb]: typo

File: spec/models/google_event_spec.rb [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    context 'given event_id that exists in db' do
+      let!(:event) { create :event }
+      let(:conference_room_id) { event.conference_room.id }
+
+      it 'confirms event' do
+        google_event.confirm(conference_room_id, event.event_id)
+
+        expect(Event.find_by(event_id: event.event_id).confirmed).to eq true
```
Comment (mjk-old) [inline spec/models/google_event_spec.rb]: What about:

``` ruby
subject {google_event.confirm(conference_room_id, event.event_id)}
it { is_expected.to change(:confirmed).from(false).to(true) }
```

File: db/migrate/20160905122739_add_default_confirmed_value.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class AddDefaultConfirmedValue < ActiveRecord::Migration[5.0]
```
Comment (mjk-old) [inline db/migrate/20160905122739_add_default_confirmed_value.rb]: Do you need this migration? The first one looks like it does the job.

File: app/models/event.rb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+class Event < ApplicationRecord
+  belongs_to :conference_room
+
+  validates_presence_of :event_id, :conference_room_id
+  validates_uniqueness_of :event_id
+
+  def confirm
+    new_record? ? self.confirmed = true : update(confirmed: true)
```
Comment (mjk-old) [inline app/models/event.rb]: Why you don't want to invoke `update` on new record?

**Comment by mjk-old** [comment]
@jtkpiotr see my comments

File: app/models/google_calendar/google_event.rb [hunk 12]
```diff
@@ -21,6 +21,12 @@ def find_by_room(time_interval, conference_room_ids)
       event_finder.by_room(time_interval, conference_room_ids)
     end
 
+    def confirm(conference_room_id, event_id)
+      conference_room = ConferenceRoom.find(conference_room_id)
```
Comment (nowNick) [inline app/models/google_calendar/google_event.rb]: Im wondering...

``` ruby
Event.find_or_create_by(conference_room_id: conference_room_id, event_id: event_id) do |event| 
  event.confirm
end
```

Will do the job? And do you really need to create an event here? Shouldn't it be created before?

**Comment by nowNick** [comment]
Just one comment from me. Other than that it looks good :)

File: app/models/google_calendar/google_event.rb (hunk 12 above)
Comment (jtkpiotr) [reply app/models/google_calendar/google_event.rb]: We want to raise RecordNotFound when conference_room_id is invalid

File: spec/models/google_event_spec.rb (hunk 7 above)
Comment (mjk-old) [reply spec/models/google_event_spec.rb]: subject { -> { google_event.confirm(conference_room_id, event_id) }  }

File: app/models/google_calendar/event_finder.rb [hunk 13]
```diff
@@ -17,6 +17,12 @@ def all(time_interval)
       list_events(time_interval, rooms)
     end
 
+    def confirmed_events(time_interval)
+      all_events = all(time_interval)
```
Comment (nowNick) [inline app/models/google_calendar/event_finder.rb]: all_google_events

File: spec/models/google_event_spec.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe GoogleCalendar::GoogleEvent do
```
Comment (nowNick) [inline spec/models/google_event_spec.rb]: delete

**Pull request merged**